src/AccountManager.mo

```diff
@@ -491,40 +497,95 @@ module {
       };
     };
 
-    /// Initiates a withdrawal by transferring tokens to another account.
-    /// Returns ICRC1 transaction index and amount of transferred tokens (fee excluded).
-    public func withdraw(to : ICRC1.Account, amount : Nat) : async* WithdrawResponse {
-
-      totalWithdrawn_ += amount;
-
-      let callResult = await* processWithdrawTransfer(to, amount);
+    func handleWithdrawal(
+      from_subaccount : ?Blob,
+      to : ICRC1.Account,
+      amount : Nat,
+      onSuccess : (txIdx : Nat) -> async* WithdrawResponse,
+      onError : (err : ProcessWithdrawTransferError) -> async* WithdrawResponse,
```

Comment:
Why do these two function have `async*` return type? Doesn't it work without the async*?

I think in the application below only bookkeeping and logging is done in the functions, no calls or throws.